addon/content/cats/app.js

```diff
@@ -1,15 +1,45 @@
 "use strict";
 
-/* global React ReactDOM require */
+/* global React ReactDOM require content addMessageListener sendAsyncMessage */
 
-/**
+const self = {
+  port: {
+    on(header, handle) {
+      addMessageListener(header, {
+        receiveMessage(message) {
+          if (message.name === header)
+            handle(message.data);
+        },
+      });
+    },
+    emit(header, data) {
+      sendAsyncMessage(header, data);
+    },
+  },
+};
+
+const sanitizeHtml = (m) => { return m; }; // disabling the sanitization. not needed. only text from the code is sent.
+
+self.port.on("FocusedCFR::load", (data) => {
+  content.addEventListener("load", () => load(data));
+});
+
+function load(data) {
+  console.log("loading");
+  let document = content.document; // eslint-disable-line no-global-assign, no-native-reassign
+  console.log(document);
+  /**
  * Shorthand for creating elements (to avoid using a JSX preprocessor)
  */
-const r = React.createElement;
+  const r = React.createElement;
+
+  // TODO bdanforth: 'require' is not importing the cats.js component correctly
+  // const Cats = require("resource://focused-cfr-shield-study-content/cats/cats.js");
```

Comment:
So things:  
1.  Require isn't going to work in firefox code *AT ALL*. 
2.  during the 'webpack' phase, 'resource' doesn't exist.
3.  During the firefox phase, require doesn't exist.

Options:
1.  You bundle all the resources in during packing using their relative (node like) require semantics.